projects/vdk-plugins/vdk-trino/tests/conftest.py

```diff
@@ -40,19 +24,80 @@ def docker_compose_file(pytestconfig):
         VDK_TRINO_USE_SSL: "False",
     },
 )
-def trino_service(docker_ip, docker_services):
+def trino_service(request):
     """Ensure that Trino service is up and responsive."""
     # os.system("echo Check open ports:")
     # os.system("ss -lntu")
-    runner = CliEntryBasedTestRunner(trino_plugin)
-
-    # give the server some time to start before checking if it is ready
-    # before adding this sleep there were intermittent fails of the CI/CD with error:
-    # requests.exceptions.ConnectionError:
-    #   ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
-    # More info: https://stackoverflow.com/questions/383738/104-connection-reset-by-peer-socket-error-or-when-does-closing-a-socket-resu
-    time.sleep(3)
-
-    docker_services.wait_until_responsive(
-        timeout=30.0, pause=0.3, check=lambda: is_responsive(runner)
-    )
+    port = int(os.environ[VDK_TRINO_PORT])
+    container = DockerContainer("trinodb/trino:408").with_bind_ports(port, port)
+    try:
+        container.start()
+        wait_for_logs_with_print(container, "SERVER STARTED", timeout=120)
+        os.environ[VDK_TRINO_HOST] = container.get_container_host_ip()
+        os.environ[VDK_TRINO_PORT] = str(container.get_exposed_port(port))
+        print(
+            f"Trino service started on port {container.get_exposed_port(port)} and host {container.get_container_host_ip()}"
+        )
+    except Exception as e:
+        print(f"Failed to start Trino service: {e}")
+        print(f"Container logs: {container.get_logs()}")
+        print(
+            "var logs: "
+            + container.exec(command=["cat", "/var/log/trino/server.log"])
+            + "\n"
+        )
+        print("java version: " + container.exec(command=["java", "-version"]) + "\n")
+        time.sleep(120)
```

Comment:
May have this value adjustable/tunable in global scope, seems it has to do with graceful start wait

Reply:
this was added temporarily. Removed now.